scripts/prepdocslib/jsonparser.py

```diff
@@ -0,0 +1,21 @@
+import json
+from typing import IO
+
+from .page import Page
+
+
+class JsonParser:
+    """
+    Concrete parser that can parse json into pages
+    """
+
+    def parse(self, content: IO):
+        offset = 0
+        data = json.loads(content.read())
+        if isinstance(data, list):
+            for i, obj in enumerate(data):
+                page_text = json.dumps(obj)
+                offset += len(page_text)
+                yield Page(i, offset, page_text)
+        elif isinstance(data, dict):
+            yield Page(1, 0, json.dumps(data))
```

Comment:
Interesting - you are using a 0-based index because of `enumerate` above but a 1-based index here for objects?

Reply:
I merged and rewrote some code, i'll double check things. The likely fix will be to make them consistent and 1 based. 0 based for array of objects, 1 based for pages (because docs start on page 1).